Traits/Reflection.php

```diff
@@ -81,4 +82,28 @@ public function getMethods() {
 	public function getProperties() {
 		return $this->_getProperties();
 	}
+
+	/**
+	 * (non-PHPdoc)
+	 * @see \Circle\UtilityBundle\Traits\Reflection::_getTraits()
+	 */
+	public function getTraits($deep = false) {
```

Comment:
also wenn du immer alle brauchs, warum dann überhaupt darüber nachdenken. Wenn ich getTraits aufrufe erwarte ich grundsätzlich, dass ich alle traits kriege, die in irgend einer form inkludiert werden. Wenn ich subsets davon will kann ich das ja nochma filtern (ich reflecte alle traits der parent klassen und diffe die beiden sets zB). Wenn das komplexer is: neue methode/neuer trait. 
Ich sehe da ehrlich gesagt keinen zweck an der stelle anzusetzen, weil das auch das single responsibility principle verletzt. Wenn da so ne konfiguration rein kann, muss die auch alles abdecken und das tut en boolean flag a de facto nich :D, das is so pseudo erweiterbar in meinen augen

abgesehen davon: 
"nur da genau zu sagen was welche Ebene ist ist nicht nachvollziehbar und daher nicht wirklich sinnvoll als API, weil kein Mensch rafft, was Ebene 1 usw bedeutet" -> es geht ja auch nich darum, dass der baum die api wäre. es geht darum den baum potenziell als datenstruktur zu benutzen, auf der die api aufbaut. Aber so wie ich das sehe, is das irrelevant hier, wg oben.